src/FastRouteRouter.php

```diff
@@ -114,12 +159,14 @@ public function match(Request $request)
         // Inject any pending routes
         $this->injectRoutes();
 
+        $dispatchData = $this->getDispatchData();
+
         $path       = $request->getUri()->getPath();
         $method     = $request->getMethod();
-        $dispatcher = $this->getDispatcher($this->router->getData());
+        $dispatcher = $this->getDispatcher($dispatchData);
         $result     = $dispatcher->dispatch($method, $path);
 
-        if ($result[0] != Dispatcher::FOUND) {
+        if ($result[0] !== Dispatcher::FOUND) {
```

Comment:
@Ocramius 
Yes, it's unrelated. $result[0] form FastRoute dispatcher dispatch call is always either `FastRoute\Dispatcher::NOT_FOUND` (int 0)
`FastRoute\Dispatcher::FOUND` (int 1)
`FastRoute\Dispatcher::METHOD_NOT_ALLOWED` (int 2)

ref: https://github.com/nikic/FastRoute/blob/master/src/Dispatcher.php